src/main/python/mlsearch/api_requester.py

```diff
@@ -312,23 +331,37 @@ def fetch_data(self) -> json:
                 self._fetch_paperwithcode()
 
             if self.params.get("source", "") == "github":
-                self._fetch_github()
+                try:
+                    self._fetch_github()
+                except BadCredentialsException:
+                    self.data["response_code"] = 400
+                    self.data["content"] = "Invalid Github developer key."
+                except RateLimitExceededException:
+                    self.data["response_code"] = 503
+                    self.data["content"] = "Access rate limitation reached."
 
             if self.params.get("source", "") == "youtube":
                 if not self._config.YOUTUBE_ORDER in self._config.VALID_YOUTUBE_ORDER:
                     self.data["response_code"] = 400
                     self.data["content"] = "Invalid Youtube Query Order."
                     return self.data
-
-                self._fetch_youtube(self.params.get("y_next_page_token", None))
+                try:
+                    self._fetch_youtube(self.params.get("y_next_page_token", None))
+                except HttpError as ex:
+                    print(str(ex))
+                    self.data["response_code"] = 400
+                    self.data[
+                        "content"
+                    ] = "Seems there is an authentication error with Youtube server."
 
             # TODO: Implement the function for Coursera. However, this function
             # may be handled by the backend server.
             if self.params.get("source", "") == "coursera":
                 pass
 
         except Exception as ex:
+            print(str(ex))
```

Comment:
Maybe we can remove this log to? 🤔

Reply:
This is a code to response to client side. Without the proper response, the client side will not know what happened at Lambda side rather than null result. Do you have a better mechanism to handle this? Please let me know.

Reply:
This is logging into Cloud Watch.